### PR TITLE
Feature/set game over script and add failure script actions

### LINKED ISSUE #203: Set game over sctipt and add failure script actions
## Description

...

## Acceptance Criteria

This story will be finished when ...

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

Add actions to set the game over script and failure scripts for narrative scripts. This allows for these scripts to be set dynamically now we have are using the LOAD_SCRIPT action.

In Ink scripts the failure states and game over scripts for the TMPH case can be added by writing:
```
INCLUDE {relative path to script/}FailureStates.ink
<- Failures.TMPH
```

## Testing instructions

Run the tests. If they pass everything should be fine.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource: 
  
closes #203

### CODE REVIEW

File: docs/constants.md [hunk 1]
```diff
@@ -139,6 +139,40 @@
...
+  - Macros
+  - OffScreenPose
+  - Ross_Cool_X
+  - Ross_X_Examination
+  - RossAnimationDemo
+  - ScreenshakeTest
+  - ShowItemDemo
+  - SpeechOptionsTest
+  - TestInk
+  - TutorialBoyAnimationDemo
```
Comment (notexplosive) [inline docs/constants.md]: Should we clean up scripts that aren't used?
Can we prevent scripts that are only used in automated testing from showing up here?

File: docs/methods/Script Loading.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Sets the game over narrative script for the currently playing narrative script
+
+
+Examples: 
+  - `&SET_GAME_OVER_SCRIPT:TMPH_GameOver`
+
+## ADD_FAILURE_SCRIPT
+Values: 
+  - [>The name of the failure script to add](../constants.md#FailureScriptAssetName)
```
Comment (notexplosive) [inline docs/methods/Script Loading.md]: stray `>` character. Maybe a typo in the docstring?

File: unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs [hunk 3]
```diff
@@ -521,12 +521,37 @@ private void RESET_PENALTIES()
...
+    private void SET_GAME_OVER_SCRIPT(GameOverScriptAssetName gameOverScriptName)
+    {
+        NarrativeGameState.NarrativeScriptPlaylist.SetGameOverScript(gameOverScriptName);
+        OnActionDone?.Invoke();
+    }
+
+    /// <summary>
+    /// Adds a failure script for the currently playing narrative script
+    /// </summary>
+    /// <param name="failureScriptName" validFiles="Assets/Resources/InkDialogueScripts/Failures/*.ink">>The name of the failure script to add</param>
```
Comment (notexplosive) [inline unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs]: found the stray `>`

**Review by notexplosive — APPROVED** [review]
Pretty soon you'll be able to just write the whole game in `.ink` files (are we already at that point?) I love it!

File: docs/constants.md (hunk 1 above)
Comment (PhoebeMitchell) [reply docs/constants.md]: Most of these need to be moved out of resources at some point. There's an issue for it here #118.

File: unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScriptPlaylist/NarrativeScriptPlaylist.cs [hunk 4]
```diff
@@ -10,36 +10,20 @@ public class NarrativeScriptPlaylist : MonoBehaviour, INarrativeScriptPlaylist
...
-    [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
+
+    public List<NarrativeScript> FailureScripts { get; } = new List<NarrativeScript>();
+    public NarrativeScript GameOverScript { get; set; }
 
     /// <summary>
     /// Call the initialise method on all narrative scripts in this playlist
     /// </summary>
     /// <param name="bgSceneList">A BGSceneList object with which to instantiate BGScenes used by narrative scripts</param>
     public void InitializeNarrativeScripts(BGSceneList bgSceneList)
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScriptPlaylist/NarrativeScriptPlaylist.cs]: Nit: The `bgSceneList` parameter no longer seems to be used on this method, so it should be safe to remove.

File: unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs [hunk 5]
```diff
@@ -521,12 +521,37 @@ private void RESET_PENALTIES()
     /// Loads a narrative script, ending the current narrative script
     /// and continuing the beginning of the loaded script
     /// </summary>
-    /// <param name="narrativeScriptName">The name of the narrative script to load</param>
+    /// <param name="narrativeScriptName" validFiles="Assets/Resources/InkDialogueScripts/*.ink">The name of the narrative script to load</param>
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs:524]: Always love seeing features I've worked on actually being used by other people in their code. 👍

**Review by ViMaSter — CHANGES_REQUESTED** [review]
Love seeing the load script/load scene idea we talked about realized and actually ending up looking pretty solid. Great work - as usual!

Requesting changes, as it seems like are a few tests are failing (aside from our usual suspect):
- The one in `unity-ggjj/Assets/Tests/PlayModeTests/Scenes/MainMenu/ViaKeyboard.cs` should be safe to ignore, but the others seem new to me.
- The failing tests in `ActionDecoderTests`, should be fairly easy to fix: The only thing missing here is valid examples for the new `*AssetName` classes inside the `ActionDecoderTests.ValidData` field for the auto-generating tests to also cover the new `SET_GAME_OVER_SCRIPT`&`ADD_FAILURE_SCRIPT` actions.

**Review by ViMaSter — APPROVED** [review]

**Pull request merged**